cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/ckan/CKANBackendImpl.java

```diff
@@ -324,6 +397,48 @@ private void createDataStore(String resId) throws CygnusRuntimeError, CygnusPers
                     + ", statusCode=" + res.getStatusCode() + ")");
         } // if else
     } // createResource
+
+    /**
+     * Creates a datastore for a given resource in CKAN.
+     * @param resId Identifies the resource whose datastore is going to be created.
+     * @param records Array list with the attributes names for being used as fields with column mode
+     * @throws Exception
+     */
+    private void createDataStoreWithFields(String resId, String records) throws CygnusRuntimeError, CygnusPersistenceError {
+        // create the CKAN request JSON
+        // CKAN types reference: http://docs.ckan.org/en/ckan-2.2/datastore.html#valid-types
+        org.json.JSONObject jsonContent = new org.json.JSONObject(records);
+        Iterator<String> keys = jsonContent.keys();
+        ArrayList <String> fields = new ArrayList<>();
+        while (keys.hasNext()) {
+            String key = keys.next();
+            fields.add(key);
+        }
+        String jsonString = "{ \"resource_id\": \"" + resId
+                + "\", \"fields\": [ ";
+        String content2="";
+        for (int i=0;i<fields.size()-1;i++){
+            content2 += "{ \"id\": \"" + fields.get(i) + "\", \"type\": \"text\"},";
+        }
+        content2 += "{ \"id\": \"" + fields.get(fields.size()-1) + "\", \"type\": \"text\"}"
+                + "], "
+                + "\"force\": \"true\" }";
+        jsonString += content2;
+
+        // create the CKAN request URL
+        String urlPath = "/api/3/action/datastore_create";
+
+        // do the CKAN request
+        JsonResponse res = doCKANRequest("POST", urlPath, jsonString);
+
+        // check the status
+        if (res.getStatusCode() == 200) {
+            LOGGER.debug("Successful datastore creation (resourceId=" + resId + ")");
+        } else {
+            throw new CygnusPersistenceError("Could not create the datastore (resId=" + resId
+                    + ", statusCode=" + res.getStatusCode() + ")");
+        } // if else
+    } // createResource
```

Comment:
Use JsonElement or gson objects to avoid creating String jsons. 

Check cygnus-ngsi columnAggregation for examples.

Reply:
This method uses the same style presented in the CKANBackendImpl.java check the method createDataStore

Reply:
That's correct. The idea is to avoid creating Jsons as Strings and use gson to do it, so the technical debt stops growing.

Reply:
I agree with @IvanHdzC . New methods should be added in the "right" way. Old method would remain using the "bad" way until somebody can fix them (tech debt).

Reply:
Ok I have made the changes in the code for using Gson objects instead of jsonString please check e611e02